webui/ocw/lib/cron.py

```diff
@@ -0,0 +1,100 @@
+from threading import Thread
+from threading import Lock
+from time import sleep
+from datetime import timedelta, datetime, timezone
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class CronJob:
```

Comment:
I think using something like this https://docs.python.org/3/library/sched.html or this https://apscheduler.readthedocs.io/en/stable/ will allow you to remove some big amount of code here

Reply:
`sched` doesn't seems to have periodical calls, so I will take a look to `apscheduler`. It looks promising, the only thing I'm missing is the information if a Job is currently running or not.
But maybe it's not that important or we just need to make a simple flag in `db.update()`

Reply:
done